### PR TITLE
chore(trace): move trace processor module

### PR DESCRIPTION
**Pull request by majorgreys**
Move `ddtrace.internal.processor.trace` and `ddtrace.internal.processor.SpanProcesssor` into the trace sub-package. The schema and peer service processors are put into their own internal subdirectories in to simplify code ownership.

Stacked after #8234 

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/_trace/processor/trace.py [hunk 1]
```diff
@@ -70,6 +69,72 @@ def process_trace(self, trace):
         pass
 
 
+@attr.s
```
Comment (christophe-papazian) [inline ddtrace/_trace/processor/trace.py:72]: why using attr dependency as we don't need to be python 2 compatible?

**Review by christophe-papazian — APPROVED** [review]

**Review by emmettbutler — COMMENTED** [review]
I can't tell for sure but it looks like this pull request depends on #8234. I'll wait to review until that one is merged. Let me know if you want feedback before then.

File: ddtrace/_trace/processor/trace.py (hunk 1 above)
Comment (majorgreys) [reply ddtrace/_trace/processor/trace.py:72]: I want to minimize code changes for this move.

File: ddtrace/_trace/processor/trace.py [hunk 2]
```diff
@@ -267,16 +332,22 @@ def shutdown(self, timeout):
...
+                telemetry.telemetry_writer._is_periodic = False
+                telemetry.telemetry_writer._enabled = True
+                # on_span_start queue span created counts in batches of 100. This ensures all remaining counts are sent
+                # before the tracer is shutdown.
+                self._queue_span_count_metrics("spans_created", "integration_name", None)
+                # on_span_finish(...) queues span finish metrics in batches of 100.
+                # This ensures all remaining counts are sent before the tracer is shutdown.
+                self._queue_span_count_metrics("spans_finished", "integration_name", None)
+                telemetry.telemetry_writer.periodic(True)
+                # Disable the telemetry writer so no events/metrics/logs are queued during process shutdown
```
Comment (ZStriker19) [inline ddtrace/_trace/processor/trace.py:349]: This is new? Why this change? Were we having issues or did you just want to add this in?

**Review by ZStriker19 — APPROVED** [review]

File: ddtrace/_trace/processor/trace.py (hunk 2 above)
Comment (majorgreys) [reply ddtrace/_trace/processor/trace.py:349]: good catch, there was a change to the original path. fixing now

**Review by ZStriker19 — APPROVED** [review]

**Review by ZStriker19 — APPROVED** [review]

**Review by shatzi — APPROVED** [review]
